docs/contentlayer.config.ts

```diff
@@ -88,6 +94,66 @@ export default makeSource({
     rehypePlugins: [
       [rehypeComponentDemo, { contentDirPath }],
       rehypeSlug,
+      // to inject the source code and other stuff inside `pre` element props
+      () => tree => {
+        visit(tree, node => {
+          if (node?.type === 'element' && node?.tagName === 'pre') {
+            const [codeEl] = node.children;
+            if (codeEl.tagName !== 'code') {
+              return;
+            }
+
+            if (codeEl.data?.meta) {
+              // Extract event from meta and pass it down the tree.
+              const regex = /event="([^"]*)"/;
+              const match = codeEl.data?.meta.match(regex);
+              if (match) {
+                node.__event__ = match ? match[1] : null;
+                codeEl.data.meta = codeEl.data.meta.replace(regex, '');
+              }
+            }
+            node.__rawString__ = codeEl.children?.[0].value;
+          }
+        });
+      },
+      [rehypeAutolinkHeadings, { behavior: 'wrap' }],
+      [
+        rehypePrettyCode,
+        {
+          getHighlighter: async () => {
```

Comment:
Why did you chose a custom theme and not a prepacked one?

Reply:
I don't understand what do you mean here

Reply:
some from rehypePrettycode

Reply:
We don't want to just copy 😄 it's fine to use a prebuild one

Reply:
you know what 
The prebuild  one looks better xD